forch/authenticator.py

```diff
@@ -163,7 +198,22 @@ def parse_args(raw_args):
             'secret': ARGS.radius_secret
         }
     }
-    AUTHENTICATOR = Authenticator(AUTH_CONFIG)
-    AUTHENTICATOR.process_auth_result()
-    if ARGS.mab:
-        AUTHENTICATOR.do_mab_request(ARGS.src_mac, ARGS.port_id)
+    MOCK_RQUERY = RQuery()
+    AUTHENTICATOR = Authenticator(AUTH_CONFIG, mock_auth_callback, MOCK_RQUERY)
+
+    # test radius query call for device placement
+    TEST_MAC = '00:aa:bb:cc:dd:ee'
+    DEV_PLACEMENT = DevicePlacement(switch='t2s2', port=1, connected=True)
+    AUTHENTICATOR.process_device_placement(TEST_MAC, DEV_PLACEMENT)
+    assert MOCK_RQUERY.query_status_updated() and MOCK_RQUERY.last_mac_query == TEST_MAC
```

Comment:
nb: this is about 2x as complicated as it needs to be, because it could have been one function that was just "last_mac_query()" that returned the value and then cleared it.

Reply:
Tried resolving the issue. Please let me know if it is fine now.

Reply:
Yup!  Perfect.